use-location.js

```diff
@@ -9,11 +9,9 @@ const eventReplaceState = "replaceState";
 export const events = [eventPopstate, eventPushState, eventReplaceState];
 
 export default ({ base = "" } = {}) => {
-  const [{ path, search }, update] = useState(() => ({
-    path: currentPathname(base),
-    search: location.search,
-  })); // @see https://reactjs.org/docs/hooks-reference.html#lazy-initial-state
-  const prevHash = useRef(path + search);
+  const [{ path }, update] = useState(() => ({ path: currentPathname(base) }));
```

Comment:
```suggestion
  const [path, update] = useState(() => currentPathname(base));
```

Would that work?

Reply:
No worries, I can fix it in the main branch.

Reply:
Turns out I was wrong here, if we change the state type to string, React won't trigger a re-render (because it's a primitive value and it doesn't change).